File: longest-palindromic-substring.cpp
#include <string>

using namespace std;

class Solution {
public:
    string longestPalindrome(string s) {
		int start = 0;
		int max_len = 0;

		for (int i=0; i<s.size(); ++i) {
			int lt = i;
			int	rt = i;
			while ((0 <= lt) && (rt < s.size()) && (s[lt] == s[rt])) {
				int length = rt - lt + 1;
				if (length > max_len) {
					start = lt;
					max_len = length;
				}
				--lt;
				++rt;
			}
			
			lt = i;
			rt = i+1;
			while ((0 <= lt) && (rt < s.size()) && (s[lt] == s[rt])) {
				int length = rt - lt + 1;
				if (length > max_len) {
					start = lt;
					max_len = length;
				}
				--lt;
				++rt;
			}
		}

		return s.substr(start, max_len);
    }
};

File: longest-palindromic-substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        start = 0
        max_len = 0

        for i in range(len(s)):
            lt, rt = i, i
            while 0 <= lt and rt < len(s) and s[lt] == s[rt]:
                length = rt - lt + 1
                if length > max_len:
                    start, max_len = lt, length
                rt += 1
                lt -= 1
                
            lt, rt = i, i+1
            while 0 <= lt and rt < len(s) and s[lt] == s[rt]:
                length = rt - lt + 1
                if length > max_len:
                    start, max_len = lt, length
                rt += 1
                lt -= 1

        return s[start:start+max_len]
